### ISSUE TITLE
wxEVT_COMMAND_AUINOTEBOOK_* events get wxAuiNotebook as EventObject instead of wxAuiTabContainer

### ISSUE THREAD

**Issue by pkulchenko** [nhold/wxWidgets#111]
All event handlers like wxEVT_COMMAND_AUINOTEBOOK_BG_DCLICK and wxEVT_COMMAND_AUINOTEBOOK_TAB_RIGHT_UP get `wxAuiNotebook` object instead of `wxAuiTabCtrl` object they were getting before. `wxAuiTabCtrl` has been "replaced" by `wxAuiTabContainer` and the events should be getting the container object instead as there may be multiple containers for the same notebook (when the tabs are split) and for proper click handling the user would need to know which of the containers is sending the event.

**Comment by mjmacleod** [comment]
Can you give an example (or even better a code snippet) of what it is you want to do in your 'proper click handling' so that I can better understand this use case?

**Comment by pkulchenko** [comment]
@mjmacleod, one example would be adding a new tab on `BG_DCLICK`. The tab needs to be added to the tab control that the click is on. Without passing the tab control (as it used to), there is no API to see what tab container the click is on.

Another example is the `AUINOTEBOOK_BUTTON` event as each tab container has its own button (when enabled).

I also use it in `TAB_RIGHT_UP` event as its `event:GetSelection()` returns the index inside the current tab control, so I need to have access to the tab control for split notebooks to get to the proper page.

I think in general it's better to pass the tab control as you can always get to the notebook from it, but the other way is not available using the public API.

**Comment by pkulchenko** [comment]
@mjmacleod, this is actually less of an issue now after you made `FindTab` method public in the new API for wxAuiManager (which is very useful), but it still applies to those events that are not tab-specific, like `BG_CLICK` and `BUTTON`. If there is a way to get to the appropriate `wxAuiTabContainer` object  using the coordinates of the event, then it may be a non-issue.

**Comment by mjmacleod** [comment]
I can't recall if there was any reason these events changed the object (better drop in compatibility with normal notebooks perhaps?) or not..

Anyway lets just change it to be backwards compatible then. I'll do a pull request when I have time, unless someone else wants to do it in the meantime.

If anyone objects to this change please say so :)

**Comment by Kinaou** [comment]
I'm on this issue and I'm checking if the `wxAuiNotebookEvent` objects contain the same info between the old and the new aui.
I realize that the event ID `wxEVT_COMMAND_AUINOTEBOOK_CANCEL_DRAG` is defined but never sent in the new aui.
In the old aui it was sent in `auibook.cpp`:

``` c++
void wxAuiTabCtrl::OnCaptureLost(wxMouseCaptureLostEvent& WXUNUSED(event))
{
    if (m_isDragging)
    {
        m_isDragging = false;

        wxAuiNotebookEvent evt(wxEVT_AUINOTEBOOK_CANCEL_DRAG, m_windowId);
        evt.SetSelection(GetIdxFromWindow(m_clickTab));
        evt.SetOldSelection(evt.GetSelection());
        evt.SetEventObject(this);
        GetEventHandler()->ProcessEvent(evt);
    }
}
```

In the new aui the equivalent method does not send this event:

``` c++
void wxAuiManager::OnCaptureLost(wxMouseCaptureLostEvent& WXUNUSED(evt))
{
    HideToolTip();

    // cancel the operation in progress, if any
    if ( m_action != actionNone )
    {
        m_action = actionNone;
        HideHint();
        delete m_actionDeadZone;
        m_actionDeadZone = NULL;
    }
}
```

Is this intentional?
If not, is there a mean to know if we are in a dragging process (like `m_isDragging`)?
=> is something like that would be OK?:

``` cpp
if (m_action == actionDragToolbarPane ||
    m_action == actionDragFloatingPane ||
    m_action == actionDragMovablePane)
{
   ...
}
```

**Comment by mjmacleod** [comment]
I think it was probably missed - we should add the event back.

**Comment by pkulchenko** [comment]
@Kinaou, good catch! I agree; I see that you already added `CANCEL_DRAG` in #125.

**Comment by pkulchenko** [comment]
Closing as this should be resolved by #125.
